pkg/transfer/local/pull.go

```diff
@@ -152,20 +160,28 @@ func (ts *localTransferService) pull(ctx context.Context, ir transfer.ImageFetch
 
 	// TODO: Should available platforms be a configuration of the service?
 	// First find suitable platforms to unpack into
-	//if unpacker, ok := is.
+	//If image storer is also an unpacker type, i.e implemented UnpackPlatforms() func
 	if iu, ok := is.(transfer.ImageUnpacker); ok {
 		unpacks := iu.UnpackPlatforms()
 		if len(unpacks) > 0 {
 			uopts := []unpack.UnpackerOpt{}
+			//Only unpack if requested unpackconfig matches default/supported unpackconfig
 			for _, u := range unpacks {
-				uopts = append(uopts, unpack.WithUnpackPlatform(u))
+				for _, du := range ts.config.UnpackConfigs {
+					if u.Platform.Match(du.Platform) && u.SnapshotterKey == du.Snapshotter {
```

Comment:
The client should be able to just pass in the platform and have `u.SnapshotterKey` be empty. In that case, it should just use whatever snapshotter is configured for the platform

Reply:
Updated in the latest commit .